Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -119,6 +128,14 @@ private Exception InvokeSubjectWithInterception()
             return actualException;
         }
 
+        private void FailIfSubjectAsyncVoid(string assertionMethodName)
+        {
+            if (Subject.GetMethodInfo().HasAttribute<AsyncStateMachineAttribute>())
```

Comment:
Not sure what FA's stance on reflection is - this is the only way I'm aware of to do this check though.